src/Core.hs

```diff
@@ -1,3 +1,12 @@
+{-|
+Module           : Core
+Description      : An extensible datatype for expressions
+
+The various instantiations of the openly recursive 'CoreF' datatype represent
+expressions in different states of expansion, with 'Core' representing
```

Comment:
As a minor quibble, I might write "with the fixed point 'Core' representing fully-expanded expressions.'